addons/website_event_sale/models/sale_order.py

```diff
@@ -78,21 +73,20 @@ def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, s
                 'event': ticket.event_id.name}
             new_qty, set_qty, add_qty = ticket.seats_available, ticket.seats_available, 0
 
-        values.update(super(sale_order, self)._cart_update(
-            cr, uid, ids, product_id, line_id, add_qty, set_qty, context, **kwargs))
+        values.update(super(SaleOrder, self)._cart_update(product_id, line_id, add_qty, set_qty, **kwargs))
 
         # removing attendees
         if ticket and new_qty < old_qty:
             attendees = Attendee.search(
-                cr, uid, [
+                [
                     ('state', '!=', 'cancel'),
-                    ('sale_order_id', '=', ids[0])
+                    ('sale_order_id', '=', self.ids[0])
                     ('event_ticket_id', '=', ticket.id)
-                ], offset=new_qty, limit=(old_qty-new_qty),
-                order='create_date asc', context=context)
-            Attendee.button_reg_cancel(cr, uid, attendees, context=context)
+                ], offset=new_qty, limit=(old_qty - new_qty),
+                order='create_date asc')
+            Attendee.button_reg_cancel(attendees)
```

Comment:
Why is that not `attendees.button_reg_cancel`? And then `Attendee` could be inlined.